### PR TITLE
(maint) Add websocket trace level logging for all events

### PR DESCRIPTION
**Pull request by rileynewton**

### CODE REVIEW

**Comment by rileynewton** [comment]
I'm not sure we actually want to do all this trace logging in production (especially for messages) - but I went ahead and bumped the CHANGELOG so we can use the new version for now at least.

File: src/puppetlabs/trapperkeeper/services/webserver/jetty10_websockets.clj [hunk 1]
```diff
@@ -77,39 +77,60 @@
...
+  (let [client-id (swap! client-count inc)
+        {:keys [on-connect on-error on-text on-close on-bytes]
          :or {on-connect (partial no-handler :on-connect)
               on-error   (partial no-handler :on-error)
               on-text    (partial no-handler :on-text)
               on-close   (partial no-handler :on-close)
               on-bytes   (partial no-handler :on-bytes)}} handlers]
     (proxy [WebSocketAdapter CertGetter ClosureLatchSyncer] []
       (onWebSocketConnect [^Session session]
+        (log/tracef "%d on-connect certname:%s uri:%s" client-id (extract-CN-from-certs x509certs) requestPath)
```
Comment (jonathannewman) [inline src/puppetlabs/trapperkeeper/services/webserver/jetty10_websockets.clj]: Can you define the value of `(extract-CN-from-certs x509certs)` in the let block so we don't have to do the work on every log line?

Comment (jonathannewman) [reply src/puppetlabs/trapperkeeper/services/webserver/jetty10_websockets.clj]: That way it is done once per instance, rather than once per log line.

Comment (rileynewton) [reply src/puppetlabs/trapperkeeper/services/webserver/jetty10_websockets.clj]: Whoops yeah

File: src/puppetlabs/trapperkeeper/services/webserver/jetty10_websockets.clj [hunk 2]
```diff
@@ -77,39 +77,60 @@
...
         (let [^WebSocketAdapter this this]
           (proxy-super onWebSocketConnect session))
-        (on-connect this))
+        (let [on-connect-result (on-connect this)]
+          (log/tracef "%d exiting on-connect certname:%s uri:%s" client-id (extract-CN-from-certs x509certs) requestPath)
```
Comment (jonathannewman) [inline src/puppetlabs/trapperkeeper/services/webserver/jetty10_websockets.clj]: I think the `exit` messages can just include the client id, and the `exiting <behavior>` and not the extra arguments (they don't change between the start of the function and the end.

Comment (jonathannewman) [reply src/puppetlabs/trapperkeeper/services/webserver/jetty10_websockets.clj]: The log lines will naturally have the thread identifier with them, so we should be able to pair them up.

File: src/puppetlabs/trapperkeeper/services/webserver/jetty10_websockets.clj (hunk 1 above)
Comment (rileynewton) [reply src/puppetlabs/trapperkeeper/services/webserver/jetty10_websockets.clj]: Updated

File: src/puppetlabs/trapperkeeper/services/webserver/jetty10_websockets.clj (hunk 2 above)
Comment (rileynewton) [reply src/puppetlabs/trapperkeeper/services/webserver/jetty10_websockets.clj]: Seems reasonable, updated

**Review by jonathannewman — APPROVED** [review]

**Pull request merged**